### DISCUSSION TITLE
Request to add Typst language support

### DISCUSSION THREAD

**Discussion by ayoubelmhamdi** [tree-sitter/tree-sitter#2153] [Grammars] [answered]
I would like to suggest adding support for [Typst](https://github.com/typst/typst), a new markup-based typesetting system for the sciences. `Typst` is designed to be an alternative to advanced tools like `LaTeX` and simpler tools like `Word` and `Google Docs`. Our goal with `Typst` is to build a typesetting tool that is highly capable and a pleasure to use.

I believe that adding support for `Typst` would be beneficial for many users who are looking for an alternative to `LaTeX`. The Typst compiler is open source, as are the libraries created for Typst such as `svg2pdf`, `biblatex`, and `pdf-writer`. I hope you will consider adding support for Typst in tree-sitter.

Please let me know if there is anything else I can provide to help make this happen. Thank you for your time and consideration.

**Comment by ahelwer** [comment]
Languages are contributed on a volunteer basis by those who want to see them supported.

**Comment by ayoubelmhamdi** [comment]
[typst.vim](https://github.com/kaarmu/typst.vim) The initial plugins are designed to enable syntax highlighting.

**Comment by uben0** [comment, marked as answer]
I wrote and maintain a grammar at https://github.com/uben0/tree-sitter-typst
